### PR TITLE
Variance (and a bit more)

### PR DESCRIPTION
**Pull request by profxj**
Progress toward calculating the variance based on sky + object models.

Turned off ampscale for armlsd (could make a parameter).
  -- For LRIS, the flat didn't span across the 2 amps in either detector

Made change to oscansub related to oscansec

Variance generated is now within 5% of LowRedux

### CODE REVIEW

File: src/arproc.py [hunk 1]
```diff
@@ -1079,22 +1078,26 @@ def rn_frame(slf, det):
...
     rnimg = np.zeros_like(slf._ampsec[det-1])
     for ii in range(slf._spect['det'][det-1]['numamplifiers']):
         amp = ii+1
         amppix = slf._ampsec[det-1] == amp
-        rnimg[amppix] = (slf._spect['det'][det-1]['ronoise'][ii]**2 +
-                         (0.5*slf._spect['det'][det-1]['gain'][ii])**2)
+        #rnimg[amppix] = (slf._spect['det'][det-1]['ronoise'][ii]**2 +
+        #                 (0.5*slf._spect['det'][det-1]['gain'][ii])**2)
+        # Assumes readnoise is given in e- not counts
+        rnimg[amppix] = slf._spect['det'][det-1]['ronoise'][ii]**2
```
Comment (rcooke-ast) [inline src/arproc.py]: Why has the gain term been removed?  This term accounts for the uncertainty in the rounding error associated with the readout. I can't see where this term is accounted for elsewhere in the code.

File: src/pypit.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # Import PYPIT routines
 import ardebug
 debug = ardebug.init()
-#debug['develop'] = True
+debug['develop'] = True
```
Comment (rcooke-ast) [inline src/pypit.py]: False for commits to master  :-)

File: src/arproc.py [hunk 3]
```diff
@@ -1202,28 +1211,34 @@ def trim(slf, file, det):
...
-    # Dark Current noise
-    dnoise = (slf._spect['det'][det-1]['darkcurr'] *
-              float(fitsdict["exptime"][idx])/3600.0)
-    # The effective read noise
-    rnoise = rn_frame(slf,det)
-    # Return
-    return np.abs(scicopy) + rnoise + dnoise
+        '''
+        if objframe is None:
+            objframe = np.zeros_like(skyframe)
```
Comment (rcooke-ast) [inline src/arproc.py]: This worries me. If objframe is None, we aren't accounting for the object photons in the variance frame. Am I missing something?

File: src/arproc.py (hunk 1 above)
Comment (profxj) [inline src/arproc.py]: Because I didn't understand it (my fault)! ;)
Returning to the way it was.

File: src/pypit.py (hunk 2 above)
Comment (profxj) [inline src/pypit.py]: Agreed.

File: src/arproc.py (hunk 3 above)
Comment (profxj) [inline src/arproc.py]: You are correct.  That is why there is a placeholder
later on (related to object extraction) for this.

Key question is what to do for boxcar extraction.  
Probably the best solution there is to stick with 
the standard variance image (i.e. use the 'raw' counts).

I'll implement that on the next PR.

Comment (rcooke-ast) [inline src/arproc.py]: Understood -- thanks for the clarification!

Agreed, let's stick with raw counts for the
boxcar.

**Comment by profxj** [comment]
merging

**Pull request merged**